multidid/src/lib.rs

```diff
@@ -0,0 +1,342 @@
+use integer_encoding::*;
+use std::str;
+use std::collections::HashMap;
+use multibase::Base;
+
+/**
+ * Multicodec Codes https://github.com/multiformats/multicodec/blob/master/table.csv
+ */
+ const MULTIDID_CODEC: u32 = 0xd1d; // did
```

Comment:
For a group of codecs, rust approach is usually to put them in an enum. That way they can be strongly typed, and easily shared by making the enum public https://github.com/3box/rust-ceramic/pull/3/files#diff-ecd9c86aee006d0568451cc27f6aba8ae777dee68cc03920d480ac80e04e5ac2R14

Reply:
For example see how mulitbase does it here https://docs.rs/multibase/latest/multibase/enum.Base.html 

Notice the `from_code` and `code` methods that allow conversion to an integer code.

A peak behind the scenes show they use a macro to implement the methods on  each variant of the enum https://docs.rs/multibase/latest/src/multibase/base.rs.html While its a bit of an advanced technique its a good practice in this case and reads well.

Reply:
tried doing this as enum now, will probably try to do as macro after, but havent gotten there yet, very verbose this way but nice to be properly typed

the did key codecs are both top level method (did:?) determinants, and method specific (did:key:?) determinants, making cross boundaries a bit, all other methods wont do this, didnt know if there was better way represent as enum here where did:key codecs need to be represented at top level and as subset

Reply:
If for some reason enums don't work, you can also create a stronger type.

```pub type Codec = u32```

then

```const MULTIDID_CODEC: Codec = 0xd1d;```

That way you can scope to only codec types, and not any u32.

Keep trying with enums though. If you still can't get it, let me know.